cmd/image-builder/repos.go

```diff
@@ -19,7 +21,41 @@ var defaultDataDirs = []string{
 	"/usr/share/image-builder",
 }
 
-var newRepoRegistry = func(dataDir string) (*reporegistry.RepoRegistry, error) {
+type repoConfig struct {
+	DataDir    string
+	ExtraRepos []string
+}
+
+func parseExtraRepo(extraRepo string) ([]rpmmd.RepoConfig, error) {
+	// XXX: we want to eventually support more repos here:
+	// - copr:@osbuild/osbuild (with full gpg retrival via the API)
+	// - https://example.com/repo/fedora-42-x86_64/ (with sslverify=True)
+	// But for now we only support local dirs
+	stat, err := os.Stat(extraRepo)
+	if err != nil {
+		return nil, fmt.Errorf("cannot add extra repository: %v", err)
+	}
+	if !stat.IsDir() {
+		return nil, fmt.Errorf("cannot add extra repository %v: must be a local directory", extraRepo)
+	}
+	extraRepo, err = filepath.Abs(extraRepo)
+	if err != nil {
+		return nil, err
+	}
+
+	checkGPG := false
+	return []rpmmd.RepoConfig{
+		{
+			Id:           extraRepo,
+			Name:         filepath.Base(extraRepo),
+			BaseURLs:     []string{fmt.Sprintf("file:%s", extraRepo)},
+			CheckGPG:     &checkGPG,
+			CheckRepoGPG: &checkGPG,
+		},
+	}, nil
+}
+
+var newRepoRegistry = func(dataDir string, extraRepos []string) (*reporegistry.RepoRegistry, error) {
```

Comment:
Why not use 
```go
func newRepoRegistry(dataDir string, extraRepos []string) (*reporegistry.RepoRegistry, error) {
...
```
?

Reply:
This is mocked in the tests so that we get the tesrepos there - I should probably add a comment. Once we https://github.com/osbuild/image-builder-cli/pull/128 is merged I can make this nicer by just mocking ` reporegistry.New()`